model/cicd/metrics.yaml

```
@@ -0,0 +1,65 @@
+groups:
+  - id: metric.cicd.pipeline.run.duration
+    type: metric
+    metric_name: cicd.pipeline.run.duration
+    brief: 'Duration of a pipeline run grouped by pipeline and result.'
+    instrument: histogram
+    unit: "s"
+    stability: experimental
+    attributes:
+      - ref: cicd.pipeline.name
+        requirement_level: required
+      - ref: cicd.pipeline.result
+        requirement_level: required
+  - id: metric.cicd.pipeline.run.executing
+    type: metric
+    metric_name: cicd.pipeline.run.executing
+    brief: 'The number of pipeline runs currently executing.'
+    instrument: updowncounter
+    unit: "{pipeline_run}"
+    stability: experimental
+    attributes:
+      - ref: cicd.pipeline.name
+        requirement_level: required
+  - id: metric.cicd.queue.latency
+    type: metric
+    metric_name: cicd.queue.latency
+    brief: 'The duration a pipeline run takes from being triggered to the start of execution.'
+    instrument: histogram
+    unit: "s"
+    stability: experimental
+    attributes:
+      - ref: cicd.pipeline.name
+        requirement_level: recommended
+  - id: metric.cicd.queue.length
+    type: metric
+    metric_name: cicd.queue.length
+    brief: 'The number of pipeline runs waiting for their start of execution.'
+    instrument: updowncounter
+    unit: "{pipeline_run}"
```

```suggestion
    unit: "{pipeline_run}"
```

 should this be the unit for queue.length?

---

What do you mean?

---

Similar metrics:
`kestrel.queued_connections` unit: `{connection}`
`kestrel.queued_requests` unit: `{request}`
`dotnet.thread_pool.queue.length` unit: `{work_item}`

Since I will change the unit for `cicd.pipeline.run.executing` from `{pipeline_run}` to `{run}` I will apply this unit name change here for consistency.

---

Done in 89f6a8f2b7ea9253495c96293b2c7a1a154d348e